demo/src/components/RangeView.js

```diff
@@ -0,0 +1,287 @@
+                        valMax: rangeValue.sliderMax,
+                        values: updatedValues,
+                    };
+                } else {
+                    return { ...rangeValue };
+                }
+            });
+
+            return {
+                ...currentFacetState,
+                rangeFacetsList: updatedRangeFacets,
+                facetChange: true
+            };
+        });
+
+        !applyMutiple && this.onApplyFilter(applyRangeFacet);
+    };
+
+    handleFacetChange = (event, facetObj, rangeFacetsListCopy, removeRangeFacet, enableApplyFilters, onFacetClick, addRangeFacet) => {
+        const facetName = facetObj.facetName;
+        removeRangeFacet({ facetName });
+        
+        let valMin=event[0],valMax = event[1];
+        let isSelected = false;
+
+        
+        this.setState((existingState) => {
+            let { rangeFacetsList } = existingState;
+            if(Object.keys(rangeFacetsList).length === 0){
+                rangeFacetsList = [...rangeFacetsListCopy]
+            }
+
+            const updatedRangeValues = rangeFacetsList.map((rangeValue) => {
+                if (rangeValue.facetName === facetName) {
+                    valMin = event[0];
+                    valMax = event[1];
+                    return {
+                        ...rangeValue,
+                        start: event[0],
+                        end: event[1]
+                    };
+
+                } else {
+                    return { ...rangeValue };
+                }
+            });
+
+            return {
+                ...existingState,
+                rangeFacetsList: updatedRangeValues,
+                facetChange: true
+            };
+        });
+        const facetObjNew = { facetName, valMin, valMax, isSelected };
+        this.setFacetValue(facetObjNew, !enableApplyFilters, onFacetClick, removeRangeFacet, addRangeFacet);
+
+    };
+
+    handleCollapseToggle = (event, rangeFacetsListCopy) => {
+        const facetName = event.target.dataset['unx_name'];
+        this.setState((existingState) => {
+            let { rangeFacetsList } = existingState;
+            if(Object.keys(rangeFacetsList).length === 0){
+                rangeFacetsList = [...rangeFacetsListCopy]
+            }
+            const updatedRangeValues = rangeFacetsList.map((rangeValue) => {
+                if (rangeValue.facetName === facetName) {
+                    return {
+                        ...rangeValue,
+                        isOpen: !rangeValue.isOpen,
+                    };
+                } else {
+                    return { ...rangeValue };
+                }
+            });
+
+            return {
+                ...existingState,
+                rangeFacetsList: updatedRangeValues,
+                facetChange: false
+            };
+        });
+    };
+
+    render(){
+        return (
+            <RangeFacets
+                transform={transform}
```

Comment:
no but we are providing it as a hook

Reply:
Do we still need to pass the transform as a prop though? The override component will receive the range facets data and it can be transformed by the integrator.

Reply:
yes you are right, the integrator can transform the range facets after receiving it but in that case there would be a difference in the internal state of the sdk and the what the integrator would be integrating